docs: Update version guidance in the FAQ (#4374)
aws/karpenter: website/content/en/docs/faq.md

## Compatibility

### Which versions of Kubernetes does Karpenter support?

> Karpenter is tested with [all currently supported EKS versions](https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html). As with all EKS supported versions, Karpenter is [supported for 14 months after first being made available](https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html#version-deprecation).

Reviewer: ```suggestion
Karpenter is tested with [all currently supported EKS versions](https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html). As with all EKS supported versions, Karpenter will [support a version for 14 months after first being made available](https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html#version-deprecation).
```

> Karpenter is tested with [all currently supported EKS versions](https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html). As with all EKS supported versions, Karpenter will [support a version for 14 months after it first being made available](https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html#version-deprecation).

Reviewer: One more grammar thing :D 
```suggestion
Karpenter is tested with [all currently supported EKS versions](https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html). As with all EKS supported versions, Karpenter will [support a version for 14 months after it's first made available](https://docs.aws.amazon.com/eks/latest/userguide/kubernetes-versions.html#version-deprecation).
```
